gazelle/python/file_parser.go

```diff
@@ -64,9 +68,27 @@ func ParseCode(code []byte) (*sitter.Node, error) {
 		return nil, err
 	}
 
-	return tree.RootNode(), nil
+	root := tree.RootNode()
+	if root.HasError() {
+		log.Printf("WARNING: failed to parse %q. The resulting BUILD target may be incorrect.", path)
+
+		verbose, envExists := os.LookupEnv("GAZELLE_VERBOSE")
+		if envExists && verbose == "1" {
+			for i := 0; i < int(root.ChildCount()); i++ {
+				child := root.Child(i)
+				if child.IsError() {
+					log.Printf("Parse error at %+v:\n%+v", child.StartPoint(), child.Content(code))
+					log.Printf("The above was parsed as: %v", child.String())
+				}
+			}
+		}
+	}
```

Comment:
nit, it is much more idiomatic to do early returns:
```suggestion
	if !root.HasError() {
		return root, nil
	}
	
	log.Printf("WARNING: failed to parse %q. The resulting BUILD target may be incorrect.", path)

	// TODO: if root.HasError, should we still return err=nil?
	verbose, envExists := os.LookupEnv("GAZELLE_VERBOSE")
	if !envExists || verbose != "1" {
		return root, nil
	}
	
	for i := 0; i < int(root.ChildCount()); i++ {
		child := root.Child(i)
		if child.IsError() {
			log.Printf("Parse error at %+v:\n%+v", child.StartPoint(), child.Content(code))
			log.Printf("The above was parsed as: %v", child.String())
		}
	}
	return root, nil
```

Reply:
Thanks! I'll update to that in a followup PR.